src/utils.py

```diff
@@ -156,35 +86,16 @@ def uin(param, default=0xDEADBEEF):
             errcho("%s [%s]: " % (param, default))
         else:
             errcho(param + ':')
-        userin = raw_input('> ')
+        userin = get_input('> ')
         if not userin or not userin.strip():
             if default != 0xDEADBEEF:
                 return default
             errcho('input is required (ctrl-c to quit)')
             continue
         return userin
 
-@osissue("refactor. one of about three implementations.")
-def getin(data, path):
-    "allows dot-path access to nested dicts"
-    path_bits = path.split('.')
-    bit, rest = path_bits[0],path_bits[1:]
-    if bit:
-        if isinstance(data, dict) and not data.has_key(bit):
-            # data not found. return None instead of throwing a fit
-            return None
-        return getin(data[bit], ".".join(rest))
-    # end of path
-    return data
-        
-def updatein(data, path, newval):
-    path_bits = path.split('.')
-    bit, rest = path_bits[0],path_bits[1:]
-    if not rest:
-        # we've come to the end of the path
-        data[bit] = newval
-        return newval
-    return updatein(data[bit], ".".join(rest), newval)
+def get_input(message):
```

Comment:
because `mock` can only patch functions on a module, not global variable; so we can mock `utils.get_input` but not `raw_input`